feat/US-003 Modificación de usuario

kuhamaven opened the pull request:
ir a la ruta Profile ahora muestra la ventana de modificación, si bien ahora esta mockeado el token, envía el token al back en el mount, y muestra al usuario logeado su perfil, ya que no se manejan ids desde el front. De momento cambié el funcionamiento de Routes, ya que estaba hardcodeado por cada ruta si se era o no admin, ahora todas cargan el admin de MockSettings.json para facilitar el testeo. Dicho mock debería cambiarse por el token y usuario real al finalizarse dicha US

kuhamaven committed:
ir a la ruta Profile ahora muestra la ventana de modificación. Routes usa el isAdmin de mockSettings para las rutas.

GustavoCarvallo commented on src/Profile/CreateOrEditProfile.css:
```diff
@@ -0,0 +1,68 @@
+.fa-user {
```
I made the comment here but the same applies to all the other .css.

Don't use the icon name (the one assigned by font awesome) to modify the icon style, because doing this would modify ALL the icons (with the same name) inside the app. 

Instead create a new custom classname. Doing this you also avoid code duplication.

For example in this case I see that your icons in CreateOrEditProfile have some common properties, so you could do something like this:

```
// on you .css
.create-edit-profile-icon {
    width: 20px;
    height: 20px;
    object-fit: contain;
    font-size: x-large;
    position: relative;
    top: 25%;
    left: 2%;
}

// on you screen
<i className="fa-user create-edit-profile-icon"/>
```

GustavoCarvallo reviewed (changes requested):
There are some things to modify but firsts seems a better idea to merge "sprint_1" into this branch, fix the conflicts, and adapt this PR to the changes made on "sprint_1".

Some of the changes made to the "sprint_1" branch are:
- GenericModal component has been refactor.
- Some styles issue have been fixed.
- Some new feature have been added.

Also all the PRs from the back-end have been merged so the integration could be done.

kuhamaven committed:
Merge branch 'sprint_1' into feat/US-003

# Conflicts:
#	src/Profile/Profile.tsx
#	src/router/Routes.tsx
#	src/utils/mocksettings.json

kuhamaven committed:
Corregido todo para funcionar con el último merge, ya esta integrado al login.

GustavoCarvallo commented on src/Profile/EditProfile.tsx:
```diff
@@ -57,72 +58,80 @@ const CreateOrEditProfile = (props: Props) => {
             passwordLengthError = true;
         }
-        if (password !== confirmPassword) {
+        if (profile.password !== confirmPassword) {
```
The user can change other data without changing the password. So you only have to validate the password requirements if the user wants to change the password.

The password validation should be done only when the user enter some text on the `password` or on the `confirm password` field.

GustavoCarvallo replied:
Stays like this for this version

GustavoCarvallo commented on src/Profile/EditProfile.tsx:
```diff
@@ -57,72 +58,80 @@ const CreateOrEditProfile = (props: Props) => {
 
     return (
-        <div className={"create-book"}>
+        <div >
```
Some recommendations:
- Always add classnames to your screens (to the first div)
- On the .css of your screen always select first you screen className to avoid applying the style to other elements in other places. 
For example, you want to change the style of an `<h1>` with the className "title" inside your login screen with className `login-screen`. To do so:

```
.login-screen .title {
    /* change your text styles here */
}
```

GustavoCarvallo commented on src/Profile/EditProfile.tsx:
```diff
@@ -57,72 +58,80 @@ const CreateOrEditProfile = (props: Props) => {
-            }
             <div>
+                <div className={"box"}>
+                    {(errors.nameError && renderError("Completar nombre")) ||
+                    (errors.lastNameError && renderError("Completar apellido")) ||
+                    (errors.passwordLengthError && renderError("La contraseña debe tener más de 6 caracteres!")) ||
+                    (errors.passwordMatchError && renderError("Las contraseñas deben coincidir!")) ||
+                    (errors.alphanumericError && renderError("La contraseña solo puede incluir letras y/o números!")) ||
+                    (errors.serverError && renderStatusError(errors.serverError))
+                    }
```
In this case, that you have some different types of error, you can create a function to render the corresponding error.

GustavoCarvallo commented on src/Profile/EditProfile.tsx:
```diff
@@ -146,10 +155,12 @@ const renderStatusError = (status: number) => {
-            <div className={"error-message"}>{message}</div>
-        </div>
+        <ErrorBox error={message} show={true}/>
+    //
+    // <div className={"error-box"}>
+    //         <div className={"error-message"}>{message}</div>
+    //     </div>
```
If the code will not be used in the future is better to get rid of it.

GustavoCarvallo commented on src/Profile/Profile.tsx:
```diff
@@ -32,8 +31,7 @@ export const EDIT = "EDIT";
-        get(`user/getLogged`, {headers: {"Content-Type": "application/json","Authorization": `Bearer ${bearString}`}})
+        get(`user/getLogged`)
             .then(res => setSelectedProfile(res))
             .catch(err => alert(err.message));},[])
```
Don't use the alert of the browser print into console instead 
`console.log(err.message)`

GustavoCarvallo commented on src/router/Routes.tsx:
```diff
@@ -13,13 +13,15 @@ import TopBar from "../TopBar/TopBar";
-
+import Login from "../Login/Login"
+import "../common/Notify.css"
+import { loggedUser, isAdmin } from "../utils/mocksettings.json";
```
When the onStart feature is merged, this mocked admin should be deleted.

GustavoCarvallo replied:
Now the onStart feature is merged on sprint_1.

GustavoCarvallo commented on src/router/Routes.tsx:
```diff
@@ -28,12 +30,10 @@ const Router = () => {
-                    <Route path={"/home"} component={Home}/>
+                    <AuthRoute path={"/book"} component={() => <ContainedComponent children={() => <Book isAdmin={isAdmin}/>} isAdmin={isAdmin} selected={0}/>}/>
+                    <Route path={"/signup"} component={SignUp}/>
+                    <AuthRoute path={'/profile'} component={ProfileView}/>
```
When the route has been selected, the item on the sidebar should be colored. Check the `/book` route where this is accomplished.

GustavoCarvallo reviewed (changes requested):
Please review the comments.

kuhamaven committed:
Merge branch 'sprint_1' into feat/US-003

kuhamaven committed:
Arreglados los bugs en Http.tsx en los post y put por el content-type.
EditProfile tiene aplicadas las correcciones solicitadas.
Se hizo el cambio en Routes para sacar el mock por los valores reales y ahora marca correctamente cuando está seleccionado Profile.

GustavoCarvallo reviewed (changes requested):
Please add the toggle icon on the password fields

kuhamaven committed:
Added toggles

GustavoCarvallo reviewed (approved):
LGTM.
Great job creating custom class names for the icons, modifying the CreateOrEditProfile (cause is not necessary in this case), and creating the error checker. :+1:
